src/Functions/FunctionsConversion.h

```diff
@@ -3188,14 +3189,40 @@ class FunctionCast final : public FunctionCastBase
         {
             return &ConvertImplGenericFromString<ColumnString>::execute;
         }
-        else
+        else if (const auto * agg_type = checkAndGetDataType<DataTypeAggregateFunction>(from_type_untyped.get()))
         {
-            if (cast_type == CastType::accurateOrNull)
-                return createToNullableColumnWrapper();
-            else
-                throw Exception(ErrorCodes::CANNOT_CONVERT_TYPE, "Conversion from {} to {} is not supported",
-                    from_type_untyped->getName(), to_type->getName());
+            if (agg_type->getFunction()->haveSameStateRepresentation(*to_type->getFunction()))
+            {
+                return [agg_type](
```

Comment:
Looks like we capture raw pointer here. Looks a bit dangerous.
Let's capture `from_type_untyped` by value and use `assert_case` in lambda.

Reply:
Ah yes. I think it's better to capture `agg_type->getFunction()` directly.